README.md

```diff
@@ -59,18 +93,38 @@ The `form_for` above will render the following HTML:
 </form>
 ```
 
+
 ## Differences between `form_for` and `form_tag`
-- The differences are subtle but extremely important to understand
-- `form_for` is essentially an advanced form helper that will yield an object (form builder) that you use to generate your form elements (text field, labels, submit button, etc)
-- `form_tag` is a lower-level form helper that simply generates a `form` element. To simply add fields to the `form_tag` block, you add form element tags, such as `text_field_tag`, `number_field_tag`, `submit_tag`, etc
-- `form_tag` makes no assumptions about what you're trying to do, and you're responsible for specifying exactly what the form is supposed to do (sending a `post` request, `patch` request, etc)
-- `form_for` handles the retrieval of values from your object model, and will also try to route the form to the appropriate action specified in the controller.
+
+Getting back to our roadtrip example, in order to make an informed decision on what route to take we need to know everything possible about both options. In like manner, in order to make a good choice for which form element to use, it's vital to understand what the differences are for each of them, the differences are subtle but extremely important to understand:
+
+* `form_for` is essentially an advanced form helper that will yield an object (form builder) that you use to generate your form elements (text field, labels, submit button, etc)
+
+* `form_tag` is a lower-level form helper that simply generates a `form` element. To simply add fields to the `form_tag` block, you add form element tags, such as `text_field_tag`, `number_field_tag`, `submit_tag`, etc
+
+* `form_tag` makes no assumptions about what you're trying to do, and you're responsible for specifying exactly what the form is supposed to do (sending a `post` request, `patch` request, etc)
+
+* `form_for` handles the retrieval of values from your object model, and will also try to route the form to the appropriate action specified in the controller.
+
+So when would you choose one over the other? Below are some real world examples:
+
+* `form_for` - this works well for forms that manage CRUD. Imagine that you have a blog posting application, a great fit for the `form_for` method would be the `Post` model. The reason is because the `Post` model would have the standard `ActiveRecord` setup, and therefore it's smart to take advantage of the functionality built into `form_for`.
+
+* `form_tag` - this works well for forms that are not directly connected with models. For example, let's say that our blog posting application has a search engine, the search form would be a great fit for using `form_tag`.
+
 
 ## Fields_for
```

Comment:
Actually can you yank out this stuff on fields_for? They haven't covered any of nested_attributes so let's just cover it later